tests/granulate_utils/test_cgroups.py

```diff
@@ -1,28 +1,231 @@
+    assert sub_cgroup is not None
+    assert sub_cgroup.has_parent_cgroup(cgroup_v1.path.name)
+
+    sub_cgroup_procs = sub_cgroup_dir / "cgroup.procs"
+    sub_cgroup.assign_process_to_cgroup(5)
+    assert sub_cgroup.get_pids_in_cgroup() == set([5])
+    assert sub_cgroup_procs.read_text() == "5"
+
+    same_cgroup = sub_cgroup.get_cgroup_in_hierarchy("dummy", sub_cgroup.path.name)
+    assert same_cgroup.path == sub_cgroup.path
+
+    parent_cgroup = sub_cgroup.get_cgroup_in_hierarchy("dummy", cgroup_v1.path.name)
+    assert parent_cgroup.path == cgroup_v1.path
+
+
+def test_cgroup_v2(tmp_path_factory: TempPathFactory):
+    tmp_dir = tmp_path_factory.mktemp("base_controller")
+    cgroup_dir = Path(tmp_dir)
+    SUB_CGROUP_NAME = "sub_cgroup"
+    sub_cgroup_dir = cgroup_dir / SUB_CGROUP_NAME
+    sub_cgroup_dir.mkdir()
+    supported_controllers = sub_cgroup_dir / "cgroup.controllers"
+    enabled_controllers = sub_cgroup_dir / "cgroup.subtree_control"
+
+    CONTROLLER_TYPE = "dummy"
+    supported_controllers.write_text(CONTROLLER_TYPE)
+    cgroup = CgroupCoreV2(cgroup_dir)
+    sub_cgroup = cgroup.get_cgroup_in_hierarchy(CONTROLLER_TYPE, SUB_CGROUP_NAME)
+    assert sub_cgroup is not None
+    assert enabled_controllers.read_text().strip() == f"+{CONTROLLER_TYPE}"
+
+    with pytest.raises(AssertionError) as exception:
+        sub_cgroup = cgroup.get_cgroup_in_hierarchy("dummy2", SUB_CGROUP_NAME)
+    assert exception.value.args[0] == "Controller not supported 'dummy2'"
+
+
+def test_get_cgroup_current_process():
+    root_path = Path("/root_path")
+    proc_fs_path = "/dummy"
+    full_path = Path("/root_path/dummy")
+    CONTROLLER_TYPE = "dummy"
+
+    with patch("granulate_utils.linux.cgroups.cgroup.get_cgroup_mount", return_value=CgroupCoreV1(root_path)):
+        with patch(
+            "granulate_utils.linux.cgroups.cgroup.read_proc_file",
+            return_value=f"1:{CONTROLLER_TYPE}:{proc_fs_path}\n".encode(),
+        ):
+            cgroup = get_cgroup_for_process(CONTROLLER_TYPE)
+            assert cgroup.path == full_path
+
+    with patch("granulate_utils.linux.cgroups.cgroup.get_cgroup_mount", return_value=CgroupCoreV2(root_path)):
+        with patch(
+            "granulate_utils.linux.cgroups.cgroup.read_proc_file",
+            return_value=f"1:{CONTROLLER_TYPE}:/fail\n0::{proc_fs_path}\n".encode(),
+        ):
+            cgroup = get_cgroup_for_process(CONTROLLER_TYPE)
+            assert cgroup.path == full_path
+
+    with pytest.raises(Exception) as exception:
+        with patch("granulate_utils.linux.cgroups.cgroup.get_cgroup_mount", return_value=CgroupCoreV2(root_path)):
+            with patch(
+                "granulate_utils.linux.cgroups.cgroup.read_proc_file",
+                return_value=f"1:{CONTROLLER_TYPE}:/fail\n".encode(),
+            ):
+                cgroup = get_cgroup_for_process(CONTROLLER_TYPE)
+    assert exception.value.args[0] == f"'{CONTROLLER_TYPE}' not found"
```

Comment:
This fails because we found only v2 mount but the process has only a single v1 controller?

Reply:
Yes